homework2/sass/header.scss

```diff
@@ -0,0 +1,343 @@
+      margin: 0 auto;
+      display: flex;
+      align-items: center;
+      justify-content: space-between;
+      padding: 20px;
+      z-index: 5;
+
+      .logo {
+        z-index: 5;
+
+        a {
+          img {
+            max-width: 92px;
+          }
+        }
+      }
+
+      .header-menu {
+        .menu-icon {
+          z-index: 5;
+          display: block;
+          position: relative;
+          width: 30px;
+          height: 20px;
+          cursor: pointer;
+
+          span, &::before, &::after {
+            left: 0;
+            position: absolute;
+            height: 15%;
+            width: 100%;
+            transition: all 0.3s ease 0s;
+            background-color: $color-primary;
+          }
+
+          &::before, &::after {
+            content: "";
+          }
+
+          &::before {
+            top: 0;
+          }
+
+          &::after {
+            bottom: 0;
+          }
+
+          span {
+            top: 50%;
+            transform: scale(1) translate(0px, -50%);
+          }
+
+          &.active {
+            span {
+              transform: scale(0) translate(0px, -50%);
+            }
+
+            &::before {
+              top: 50%;
+              transform: rotate(-45deg) translate(0px -50%);
+              -webkit-transform: rotate(-45deg)
+            }
+
+            &::after {
+              top: 50%;
+              transform: rotate(45deg) translate(0px 50%);
+              -webkit-transform: rotate(45deg)
+            }
+          }
+        }
+
+        nav {
+          position: fixed;
+          top: 0;
+          left: -100%;
+          width: 100%;
+          height: 100%;
+          background-color: $color-primary;
+          padding-top: 92px;
+          transition: left 0.3s ease 0s;
+          overflow: auto;
+          z-index: 4;
+
+          ul {
+            li {
+              list-style-type: none;
+              display: flex;
+              flex-direction: column;
+              align-items: center;
+
+              a {
+                text-decoration: none;
+                text-transform: uppercase;
+                @include fontProp(12px, 24px, $color-link);
+                letter-spacing: 2px;
+                padding: 8px;
+
+                &:hover {
+                  @include fontProp(14px, 24px, $color-link-hover);
+                }
+
+                &.current {
+                  padding-bottom: 3px;
+                  display: flex;
+                  flex-direction: column;
+                  align-items: center;
+
+                  &:after {
+                    content: "";
+                    width: 40px;
+                    height: 2px;
+                    background-color: $color-link;
+                    display: block;
+                    text-align: center;
+                    transition: all 0.3s ease 0s;
+                  }
+                }
+              }
+
+
+              &.get-skout-btn, &.login-btn {
```

Comment:
you don't need to create a new class for each button, the buttons on the page has only two states, so in most cases you just need a general class for common styles and uniqu class for each state which is just change a color , background etc.

Reply:
Done